e2etest/e2etest_suite_test.go

```diff
@@ -70,3 +83,218 @@ func TestE2E(t *testing.T) {
+	}
+
+	// Run the BGP router in a container.
+	volume := fmt.Sprintf("%s:/etc/frr", frrConfigTestDir)
+	out, err := runCommand(false, "docker", "run", "-d", "--privileged", "--network", "kind", "--rm", "--name", consts.FRRContainerName, "--volume", volume, "frrouting/frr:v7.5.1")
+	if err != nil {
+		return fmt.Errorf("Failed to start frr container. %s: %s", out, err)
+	}
+
+	return nil
+}
+
+// Try to get 10 IP addresses after the last used address in the kind network subnet.
+func getL2AddressPoolRange() ([]string, []string, error) {
+	// Get the used ipv4 and ipv6 addresses.
+	networkSubnet, err := runCommand(false, "docker", "network", "inspect", "-f", "{{range .IPAM.Config}}{{.Subnet}} {{end}}", "kind")
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get subnet network info: %s", err)
+	}
+	networkSubnet = strings.TrimSpace(networkSubnet)
+
+	usedIpv4, err := runCommand(false, "docker", "network", "inspect", "-f", "{{range .Containers}}{{.IPv4Address}} {{end}}", "kind")
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get ipv4 addresses: %s", err)
+	}
+	usedIpv4 = strings.TrimSpace(usedIpv4)
+
+	usedIpv6, err := runCommand(false, "docker", "network", "inspect", "-f", "{{range .Containers}}{{.IPv6Address}} {{end}}", "kind")
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get ipv6 addresses: %s", err)
+	}
+	usedIpv6 = strings.TrimSpace(usedIpv6)
+
+	// Get the address pool range.
+	ipv4Range := make([]string, 2)
+	ipv6Range := make([]string, 2)
+	for _, network := range strings.Split(networkSubnet, " ") {
+		// Check is IPv4 network
+		if strings.Count(network, ":") < 2 {
+			// Sort the ips to get the last used ip address.
+			usedIpv4List := sortIPs(strings.Split(usedIpv4, " "))
+			lastusedIP := usedIpv4List[len(usedIpv4List)-1]
+			nextIP, err := incrementIPByValue(lastusedIP, network, 1)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get first free ipv4 address: %s", err)
+			}
+			// Get a range of 10 unused ipv4 addresses.
+			ipv4Range, err = getIPsRange(nextIP, network, 10)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get ipv4 range: %s", err)
+			}
+		} else {
+			// Sort the ips to get the last used ip address.
+			usedIpv6List := sortIPs(strings.Split(usedIpv6, " "))
+			lastusedIP := usedIpv6List[len(usedIpv6List)-1]
+			nextIP, err := incrementIPByValue(lastusedIP, network, 1)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get first free ipv6 address: %s", err)
+			}
+			// Get a range of 10 unused ipv6 addresses.
+			ipv6Range, err = getIPsRange(nextIP, network, 10)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get ipv6 range: %s", err)
+			}
+		}
+	}
+
+	return ipv4Range, ipv6Range, nil
+}
+
+// Configure MetalLB for testing.
+// Identify the unused network address range and use it in configmap.
+func configureMetalLB(cs clientset.Interface) error {
+	err := configureFRR(cs)
```

Comment:
Will we need to reconfigure FRR? If we do, we may want to move `configureFRR()` out of this function.